src/gui.zig

```diff
@@ -59,6 +59,27 @@ pub fn init(allocator: std.mem.Allocator) void {
         }
     }
 }
+/// Allows sharing a context across static/DLL boundaries. This is useful for
+/// hot-reloading mechanisms which rely on shared libraries.
+/// See "CONTEXT AND MEMORY ALLOCATORS" section of ImGui docs.
+pub fn initWithExistingContext(allocator: std.mem.Allocator, ctx: Context) void {
+    mem_allocator = allocator;
+    mem_allocations = std.AutoHashMap(usize, usize).init(allocator);
+    mem_allocations.?.ensureTotalCapacity(32) catch @panic("zgui: out of memory");
+    zguiSetAllocatorFunctions(zguiMemAlloc, zguiMemFree);
+
+    zguiSetCurrentContext(ctx);
+
+    temp_buffer = std.ArrayList(u8).init(allocator);
+    temp_buffer.?.resize(3 * 1024 + 1) catch unreachable;
```

Comment:
I see this has been copied from init fn but I don't like it. Just making a note.